### ISSUE TITLE
ISY Binding: Scene status doesn't update when manually operating switches

### ISSUE THREAD

**Issue by ve7gel** [QuailAutomation/openhab2-addons#16]
Another wrinkle for you Craig.  When manually operating any of the wall switches, the associated scene status doesn't update in any of the UIs.  The switch thing works normally, and the scene function works correctly when activated in the panels.  Here is the trace from manually actually device 2A 40 99, which is a member of the scene in question:

`2017-06-12 08:21:33.410 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: _1, Action: 3, Node: 
2017-06-12 08:21:33.420 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2034" sid="uuid:2040"><control>_1</control><action>3</action><node></node><eventInfo>[   n001_cams]      DON   0</eventInfo></Event>
2017-06-12 08:21:33.438 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: _1, Action: 3, Node: 
2017-06-12 08:21:39.268 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2035" sid="uuid:2040"><control>DON</control><action>0</action><node>2A 40 99 1</node><eventInfo></eventInfo></Event>
2017-06-12 08:21:39.330 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: DON, Action: 0, Node: 2A 40 99 1
2017-06-12 08:21:39.344 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - find thing handler for address: 2A 40 99 1
2017-06-12 08:21:39.347 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - Find thing for address: 2A 40 99
2017-06-12 08:21:39.350 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - address: 2A 40 99
2017-06-12 08:21:39.353 [DEBUG] [binding.isy.handler.IsyDeviceHandler] - handleUpdate called, parameters: DON:0:2A 40 99 1:
2017-06-12 08:21:39.414 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2036" sid="uuid:2040"><control>ST</control><action uom="100" prec="0">255</action><node>2A 40 99 1</node><eventInfo></eventInfo><fmtAct>100%</fmtAct></Event>
2017-06-12 08:21:39.435 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: ST, Action: 255, Node: 2A 40 99 1
2017-06-12 08:21:39.447 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - find thing handler for address: 2A 40 99 1
2017-06-12 08:21:39.450 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - Find thing for address: 2A 40 99
2017-06-12 08:21:39.452 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - address: 2A 40 99
2017-06-12 08:21:39.455 [DEBUG] [binding.isy.handler.IsyDeviceHandler] - handleUpdate called, parameters: ST:255:2A 40 99 1:
2017-06-12 08:21:39.521 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2037" sid="uuid:2040"><control>ST</control><action uom="100" prec="0">255</action><node>2A 38 92 1</node><eventInfo></eventInfo><fmtAct>100%</fmtAct></Event>
2017-06-12 08:21:39.560 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: ST, Action: 255, Node: 2A 38 92 1
2017-06-12 08:21:39.563 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - find thing handler for address: 2A 38 92 1
2017-06-12 08:21:39.566 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - Find thing for address: 2A 38 92
2017-06-12 08:21:39.568 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - No thing discovered for address: 2A 38 92 1
2017-06-12 08:21:39.571 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2038" sid="uuid:2040"><control>_1</control><action>3</action><node></node><eventInfo>[  2A 40 99 1]      DON   0</eventInfo></Event>
2017-06-12 08:21:39.610 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: _1, Action: 3, Node: 
2017-06-12 08:21:39.613 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2039" sid="uuid:2040"><control>_1</control><action>3</action><node></node><eventInfo>[  2A 40 99 1]       ST 255 (uom=100 prec=0)</eventInfo></Event>
2017-06-12 08:21:39.644 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: _1, Action: 3, Node: 
2017-06-12 08:21:39.646 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2040" sid="uuid:2040"><control>_1</control><action>0</action><node></node><eventInfo><id>2E</id><on /><nr /><r>170612 08:21:40</r><f>170612 08:21:40</f><s>31</s></eventInfo></Event>
2017-06-12 08:21:39.677 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: _1, Action: 0, Node: 
2017-06-12 08:21:39.679 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2041" sid="uuid:2040"><control>_1</control><action>0</action><node></node><eventInfo><id>2D</id><on /><nr /><r>170612 08:21:40</r><f>170612 08:17:52</f><s>22</s></eventInfo></Event>
2017-06-12 08:21:39.699 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: _1, Action: 0, Node: 
2017-06-12 08:21:40.418 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2042" sid="uuid:2040"><control>_1</control><action>0</action><node></node><eventInfo><id>2D</id><on /><nr /><r>170612 08:21:40</r><f>170612 08:21:40</f><s>21</s></eventInfo></Event>
2017-06-12 08:21:40.437 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: _1, Action: 0, Node: 
2017-06-12 08:21:40.440 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2043" sid="uuid:2040"><control>ST</control><action uom="100" prec="0">255</action><node>2A 96 64 3</node><eventInfo></eventInfo><fmtAct>100%</fmtAct></Event>
2017-06-12 08:21:40.455 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: ST, Action: 255, Node: 2A 96 64 3
2017-06-12 08:21:40.457 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - find thing handler for address: 2A 96 64 3
2017-06-12 08:21:40.459 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - Find thing for address: 2A 96 64
2017-06-12 08:21:40.472 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - No thing discovered for address: 2A 96 64 3
2017-06-12 08:21:40.474 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2044" sid="uuid:2040"><control>_1</control><action>3</action><node></node><eventInfo>[  2A 38 92 1]       ST 255 (uom=100 prec=0)</eventInfo></Event>
2017-06-12 08:21:40.488 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: _1, Action: 3, Node: 
2017-06-12 08:21:40.491 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2045" sid="uuid:2040"><control>_1</control><action>3</action><node></node><eventInfo>[  2A 96 64 3]       ST 255 (uom=100 prec=0)</eventInfo></Event>
2017-06-12 08:21:40.506 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: _1, Action: 3, Node: 
2017-06-12 08:21:41.983 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2046" sid="uuid:2040"><control>DOF</control><action>0</action><node>2A 40 99 1</node><eventInfo></eventInfo></Event>
2017-06-12 08:21:43.619 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: DOF, Action: 0, Node: 2A 40 99 1
2017-06-12 08:21:43.621 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - find thing handler for address: 2A 40 99 1
2017-06-12 08:21:43.622 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - Find thing for address: 2A 40 99
2017-06-12 08:21:43.624 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - address: 2A 40 99
2017-06-12 08:21:43.625 [DEBUG] [binding.isy.handler.IsyDeviceHandler] - handleUpdate called, parameters: DOF:0:2A 40 99 1:
2017-06-12 08:21:43.687 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2047" sid="uuid:2040"><control>ST</control><action uom="100" prec="0">0</action><node>2A 40 99 1</node><eventInfo></eventInfo><fmtAct>0%</fmtAct></Event>
2017-06-12 08:21:43.704 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: ST, Action: 0, Node: 2A 40 99 1
2017-06-12 08:21:43.705 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - find thing handler for address: 2A 40 99 1
2017-06-12 08:21:43.707 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - Find thing for address: 2A 40 99
2017-06-12 08:21:43.708 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - address: 2A 40 99
2017-06-12 08:21:43.710 [DEBUG] [binding.isy.handler.IsyDeviceHandler] - handleUpdate called, parameters: ST:0:2A 40 99 1:
2017-06-12 08:21:43.771 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2048" sid="uuid:2040"><control>ST</control><action uom="100" prec="0">0</action><node>2A 38 92 1</node><eventInfo></eventInfo><fmtAct>0%</fmtAct></Event>
2017-06-12 08:21:43.793 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: ST, Action: 0, Node: 2A 38 92 1
2017-06-12 08:21:43.794 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - find thing handler for address: 2A 38 92 1
2017-06-12 08:21:43.795 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - Find thing for address: 2A 38 92
2017-06-12 08:21:43.797 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - No thing discovered for address: 2A 38 92 1
2017-06-12 08:21:43.799 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2049" sid="uuid:2040"><control>_1</control><action>3</action><node></node><eventInfo>[  2A 40 99 1]      DOF   0</eventInfo></Event>
2017-06-12 08:21:43.812 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: _1, Action: 3, Node: 
2017-06-12 08:21:43.814 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2050" sid="uuid:2040"><control>_1</control><action>3</action><node></node><eventInfo>[  2A 40 99 1]       ST   0 (uom=100 prec=0)</eventInfo></Event>
2017-06-12 08:21:43.827 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: _1, Action: 3, Node: 
2017-06-12 08:21:43.828 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2051" sid="uuid:2040"><control>_1</control><action>0</action><node></node><eventInfo><id>2E</id><on /><nr /><r>170612 08:21:43</r><f>170612 08:21:40</f><s>22</s></eventInfo></Event>
2017-06-12 08:21:43.843 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: _1, Action: 0, Node: 
2017-06-12 08:21:43.845 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2052" sid="uuid:2040"><control>_1</control><action>0</action><node></node><eventInfo><id>2D</id><on /><nr /><r>170612 08:21:43</r><f>170612 08:21:43</f><s>31</s></eventInfo></Event>
2017-06-12 08:21:43.861 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: _1, Action: 0, Node: 
2017-06-12 08:21:43.863 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2053" sid="uuid:2040"><control>_1</control><action>0</action><node></node><eventInfo><id>2E</id><on /><nr /><r>170612 08:21:43</r><f>170612 08:21:43</f><s>21</s></eventInfo></Event>
2017-06-12 08:21:43.879 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: _1, Action: 0, Node: 
2017-06-12 08:21:43.881 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2054" sid="uuid:2040"><control>ST</control><action uom="100" prec="0">0</action><node>2A 96 64 3</node><eventInfo></eventInfo><fmtAct>0%</fmtAct></Event>
2017-06-12 08:21:43.929 [DEBUG] [sy.internal.IsyWebSocketSubscription] - Control: ST, Action: 0, Node: 2A 96 64 3
2017-06-12 08:21:43.930 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - find thing handler for address: 2A 96 64 3
2017-06-12 08:21:43.931 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - Find thing for address: 2A 96 64
2017-06-12 08:21:43.933 [DEBUG] [binding.isy.handler.IsyBridgeHandler] - No thing discovered for address: 2A 96 64 3
2017-06-12 08:21:43.934 [TRACE] [sy.internal.IsyWebSocketSubscription] - Parsing message: <?xml version="1.0"?><Event seqnum="2055" sid="uuid:2040"><control>_1</control><action>3</action><node></node><eventInfo>[  2A 38 92 1]       ST   0 (uom=100 prec=0)</eventInfo></Event`

Cheers,
Gordon

**Comment by craigham** [comment]
yes, I have struggled with how to handle scenes.  When a scene button is clicked, it does initiate the scene, and the lights do their thing.  But what then happens when someone manually adjusts a light which is in that scene?  isy doesn't keep track of things like that.  I am not at my isy setup right now, but when I have a insteon switch with those scene toggles, it makes as much sense to me to not leave them lit up after pressing.  Leaving one of those buttons lit up on the wall, really just shows what that last scene that was initiated was, but that could have been anytime ago if you adjust all your lights manually too.

So, bottom line, not sure what the UI should show in openhab, open to suggestions.

**Comment by mitchmitchell** [comment]
Craig -- I had thought about this, in the context of keypadlinc leds displaying scene status.
It really comes down to a scene partially on or off -- but one way to look at it is to consider what the user wants when activating the scene controller again -- either you can turn the scene on or off.  I'd say, if the scene is turned on and one light is turned off (for example), pressing the button again turns the scene off.  This would be consistent with the behavior if no lights were manually changed.  The only place where this falls down, is if *all* the items in the scene were turned off.

I finally talked myself into considering the scene state changed only if the scene itself was changed via one of its controllers -- manually changing state of devices in the scene would not affect the scene state.  its not really right, but it has the advantage of being consistent with the behavior if no devices were manually changed.

I think its a question where you can't really define a 'correct' answer, the best you can hope for is to be consistent so you don't surprise the user.

**Comment by ve7gel** [comment]
It occurs to me that I should have been clearer.  The scenes I'm referring to are simple ones that are a group of 2477S's that are only on/off switches.  These work in place of the classic "3-way" switch scenario and will control a light or two from 2 or 3 different locations.  The other switches all track the state of the one that is actuated (one is turned off, they all turn off, as does the light - and the scene state changes from on to off in the ISY). Some of these will include a button on keypadlinc too, but again only as an on/off toggle switch.  I can see where multi-level scenes could be difficult but the simple on/off state could perhaps be reported.  Thoughts?

**Comment by craigham** [comment]
I'll think about what I should do.  I'm going to have limited bandwidth for
the next couple weeks, so might take some time.

**Comment by mitchmitchell** [comment]
If the ISY is already tracking / changing the scene state then that would be relatively straightforward at least as far as the use case is concerned wouldn't it?

**Comment by ve7gel** [comment]
Yes, all three of the switches is set up as a controller/responder, and are all linked together.  One of them acts as the "Master" - i.e. the one that actually switches the load, the others I guess could be termed "slaves" with no load connection, though they do track the state of the scene - ie if one is switched on, they all show on, as does the load.  Conversely, if one is off they are all showing off as is the load.  The isy tracks this scene state in one of two states - either on or off.   When I said manually, I meant the scene was turned off manually at one of the switches, which turns off the entire scene, rather than by a program or controller through the isy.  In any case the isy does track the state of the scene in this case.

 I don't have any of the scenario described in Mitch's last paragraph.

**Comment by robertw** [comment]
Just my 2 cents I would keep it simple scene on and off most of the time scenes are called from programs maybe a function called reset scene bring it back to the scene state would be all need .......

**Comment by craigham** [comment]
I think @ve7gel is talking about something different than you guys and scenes.

@ve7gel If I am not mistaken, when you press a paddle on one of your switches which are part of a 3-way like setup, the other switches should change to reflect the state of the load.  Could you click one of the buttons again, and send me the event.log portion?

**Comment by craigham** [comment]
As for 'Scenes' within the ISY, I don't see how we can track their state.  As far as I know there are no messages sent by the Isy which are related to scenes.  It just sends out the state changes of the various responders to the scene.  I think from within openhab the best we can do is either toggle on or toggle off a scene.  That really is all that is supported within Isy too.  Unless I am mistaken you can't get the state of a scene, just turn it on or off.  Am I missing something ?

**Comment by ve7gel** [comment]
Hi Craig, yes, the other switches should reflect the state of the one that was activated.   I'll send you the  log but it probably won't be until Thursday as I'm away from home at the moment.

**Comment by ve7gel** [comment]
Hi Craig, here is the capture from the events.log for one of the 3-way switch groups.  Not sure there's much you can do with it since there's no indication they are members of a scene.

Thanks,
Gordon

`2017-07-20 15:21:00.806 [ItemStateChangedEvent     ] - isy_switch_f75c62c0_2A_43_C3_paddleaction changed from DOF to DON
2017-07-20 15:21:00.922 [ItemStateChangedEvent     ] - isy_switch_f75c62c0_2A_53_A_state changed from OFF to ON
2017-07-20 15:21:01.031 [ItemStateChangedEvent     ] - isy_switch_f75c62c0_2A_43_14_state changed from OFF to ON
2017-07-20 15:21:01.171 [ItemStateChangedEvent     ] - isy_switch_f75c62c0_2A_43_C3_state changed from OFF to ON
2017-07-20 15:21:03.097 [ItemStateChangedEvent     ] - isy_switch_f75c62c0_2A_43_C3_paddleaction changed from DON to DOF
2017-07-20 15:21:03.182 [ItemStateChangedEvent     ] - isy_switch_f75c62c0_2A_53_A_state changed from ON to OFF
2017-07-20 15:21:03.314 [ItemStateChangedEvent     ] - isy_switch_f75c62c0_2A_43_14_state changed from ON to OFF
2017-07-20 15:21:03.397 [ItemStateChangedEvent     ] - isy_switch_f75c62c0_2A_43_C3_state changed from ON to OFF
`
